cpp-utils/include/ot_utils/core/filters/sma.hpp

```diff
@@ -0,0 +1,68 @@
+#pragma once
+
+#include <limits>
+
+#include "ot_utils/core/fixed_point.hpp"
+
+// Simple Moving Average Algorithm
+// A finite impulse response (FIR) or 'windowed' filter
+
+// Use this filter when you have a finite window
+// of data that is particularly noisy.
+namespace ot_utils {
+
+namespace filters {
+
+template <class IntegerValue, int WindowSize = 10, int Radix = 8>
+class SimpleMovingAverage {
```

Comment:
let's add a `ready() -> bool` method that a caller can use to check if values are ready. or make `compute` return an optional or something. right now, you can't really tell when the average isn't actually running without doing part of the average's job (deciding what the window is and how it works) and since this is going to be used to reduce noise, you're expecting it to, well, have less noise. But for the first 10 samples it's just exactly the input so you'd make bad decisions.